pkg/nodejoiner/addnodes.go

```diff
@@ -49,3 +72,72 @@ func saveParams(directory, kubeConfig string) error {
 	}
 	return params.Save(directory)
 }
+
+// InitClients initiates a client.
+func initk8sClient(kubeconfig string) (*kubernetes.Clientset, error) {
+	var config *rest.Config
+	var err error
+	if kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	k8sclientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return k8sclientset, err
+}
+
+func getOrCreateAuthToken(k8sclientset *kubernetes.Clientset, directory string) (string, error) {
```

Comment:
I think that this code that takes care of creating the configmap should be moved directly in the `AuthConfig` asset, conditionally based on the `workflow.AgentWorkflow` asset: if the workflow is of type `workflow.AgentWorkflowTypeAddNodes`, then the config map could be directly created within the target cluster, since it means we're in the add nodes image creation workflow.

Reply:
I think there would be challenges to get the `kubeconfigPath` while generating the AuthConfig asset.